src/beatkeeper.rs

```diff
@@ -469,6 +502,16 @@ impl BeatKeeper {
             }
         }
 
+        // Neue Schleife für die .DAT-Dateipfade
+        for (i, anlz_file) in rb.get_anlz_files()?.iter().enumerate(){
```

Comment:
This should be run within the `if slow_update {...`.  Finding the track name and ANLZ path is not time critical like the beat tracking (which runs as ~120Hz), so we can run that just a few times per second. Reading strings is also more taxing than just reading a byte or a float because of two reasons:
- There's more data being read
- Every read required the pointer chain to be re-followed, as strings are heap allocated (they move around more and are easier to lose), which also requires several reads. The other values used you only need to find once at startup and then they stick around at the same spot in memory.

I have however not benchmarked this so it's mostly just guesswork and good practice, I don't actually know if it matters